### DISCUSSION TITLE
Using own data on Time Series Online Learning is failed

### DISCUSSION THREAD

**Discussion by DanceGhost** [pycaret/pycaret#2048] [Q&A] [answered]
I want to using Time Series Online Learning,the code "updated_model = final_model.update(y=y_for_later, update_params=True)" running is errored, the error is 'NotImplementedError: <class 'pandas.core.indexes.base.Index'> is not supported for y, use one of (<class 'pandas.core.indexes.numeric.Int64Index'>, <class 'pandas.core.indexes.range.RangeIndex'>, <class 'pandas.core.indexes.period.PeriodIndex'>, <class 'pandas.core.indexes.datetimes.DatetimeIndex'>) instead.',but my data type is <class 'pandas.core.indexes.datetimes.DatetimeIndex'> .Can you help me solve it, thanks!

**Comment by ngupta23** [comment, marked as answer]
You will need to post a completely reproducible example (with data and code) so we can reproduce the error

**Reply by DanceGhost** [reply]
I get it ,but i executed ''save_model(updated_model, model_path)" is errored when the "updated_model = final_model.update(y=y_for_later, update_params=True)" was executed. the error was "ValueError: _CURRENT_EXPERIMENT global variable is not set. Please run setup() first." if i not execute "setup", just execute "load_model“ before executing "updated_model = final_model.update(y=y_for_later, update_params=True)" is not allowed ? how to solve it?

**Reply by ngupta23** [reply]
I am sorry @DanceGhost . I can not tell you what is wrong from your message. Please consider posting a reproducible example (data + code) so I can reproduce the error and debug. Thanks!

Please refer to this for details: https://stackoverflow.com/help/minimal-reproducible-example

**Reply by DanceGhost** [reply]
[https://github.com/pycaret/pycaret/discussions/1767](url), I just execute the finally code 
`#### Load data ----
from pycaret.datasets import get_data
y = get_data("airline")

#### Reserve data for online learning (last data point) ----
y_for_later = y[-1:]

#### Original model trained on all but the last data point ----
y = y[:-1]

#### Setup experiment ----
from pycaret.internal.pycaret_experiment import TimeSeriesExperiment
exp = TimeSeriesExperiment()
exp.setup(data=y, fh=12, session_id=42)

#### Initial Training ----
model = exp.create_model("ets")

final_model = exp.finalize_model(model)

#### Perform online learning with additional data point and make new prediction ----
updated_model = final_model.update(y=y_for_later, update_params=True)

from pycaret.time_series import *
_ = save_model(updated_model, '1')`
![image](https://user-images.githubusercontent.com/37367851/150451473-74a75a45-a0c8-4402-b6ab-8c6a562a1c5f.png)

**Reply by ngupta23** [reply]
It is because you switched from OOP interface to Functional Interface before saving. You can not do that. Stick to any one.

So instead of this

```python
from pycaret.time_series import *
_ = save_model(updated_model, '1')`
```
use this instead

```python
_ = exp.save_model(updated_model, '1')`
```

**Reply by DanceGhost** [reply]
I  get it,thank you!
